### PR TITLE
Added dfs.cpp

### PR DESCRIPTION
**Pull request by ajayaravindh**
The file dfs.cpp does a Depth First Search in a graph. It is mainly iterative and uses a stack, but uses recursion to handle disconnected graph case.

### CODE REVIEW

File: graph-algos/DFS/dfs.cpp [hunk 1]
```diff
@@ -1,9 +1,125 @@
...
+		}
+	}
+	// ------ End of core logic
+	cout << "\n";
+}
+
+int main()
+{
+	int n, src, dstn, chc;
+	cout << "Enter the number of edges in the graph\n";
```
Comment (22shubh22) [inline graph-algos/DFS/dfs.cpp]: actually n is number of nodes and not  number of edges.

**Review by 22shubh22 — CHANGES_REQUESTED** [review]
change number of edges to number of nodes in line 78

File: graph-algos/DFS/dfs.cpp (hunk 1 above)
Comment (ajayaravindh) [reply graph-algos/DFS/dfs.cpp]: Yup

**Comment by ajayaravindh** [comment]
I have made the necessary change. Ready for review.

**Review by 22shubh22 — APPROVED** [review]

**Comment by khanchi97** [comment]
thanks @22shubh22 👍

**Pull request merged**